cdap-ui/app/cdap/components/Wrangler/WranglerRightPanel/ChartPanel/index.js

```diff
@@ -0,0 +1,226 @@
+
+    this.state = {
+      columns: headersList,
+      isExpanded: true,
+      yValue: headersList[0]
+    };
+
+    this.onExpandClick = this.onExpandClick.bind(this);
+    this.addYValue = this.addYValue.bind(this);
+  }
+
+  componentWillMount() {
+    WranglerStore.subscribe(() => {
+      this.setState({columns: WranglerStore.getState().wrangler.headersList});
+    });
+  }
+
+  onExpandClick() {
+    this.setState({isExpanded: !this.state.isExpanded});
+  }
+
+  onXChange(event, graph) {
+    graph.x = event.target.value;
+
+    WranglerStore.dispatch({
+      type: WranglerActions.editChart,
+      payload: {
+        chart: graph
+      }
+    });
+  }
+
+  removeYValue(value) {
+    let newGraphSpec = Object.assign({}, this.props.chart);
+
+    newGraphSpec.y.splice(newGraphSpec.y.indexOf(value), 1);
+
+    WranglerStore.dispatch({
+      type: WranglerActions.editChart,
+      payload: {
+        chart: newGraphSpec
+      }
+    });
+  }
+
+  addYValue() {
+    const columnToAdd = this.state.yValue;
+    if (!columnToAdd) { return; }
+
+    let newGraphSpec = Object.assign({}, this.props.chart);
+    newGraphSpec.y.push(columnToAdd);
+
+    let availableYOptions = xor(this.state.columns, newGraphSpec.y);
+
+    this.setState({
+      yValue: availableYOptions.length ? availableYOptions[0] : ''
+    });
+
+    WranglerStore.dispatch({
+      type: WranglerActions.editChart,
+      payload: {
+        chart: newGraphSpec
+      }
+    });
+  }
+
+  renderPanelContent() {
+    if (!this.state.isExpanded) { return null; }
+
+    const graph = this.props.chart;
+
+    let chart;
+    if (graph.x && graph.y.length > 0) {
+      chart = <Charts spec={graph} />;
+    }
+
+    let yOptions = xor(this.state.columns, graph.y);
+
+    let ySelect;
+    if (yOptions.length) {
+      ySelect = (
+        <div className="y-axis-selector">
+          <div className="y-value-selector">
+            <select
+              className="form-control"
+              value={this.state.yValue}
+              onChange={e => this.setState({yValue: e.target.value})}
+            >
+              {yOptions.map((column) => {
+                return (
+                  <option
+                    key={column}
+                    value={column}
+                  >
+                    {column}
+                  </option>
+                );
+              })}
+            </select>
+          </div>
+
+          <div className="add-y-value-button">
+            <div className="y-axis-add-button text-center">
+              <span
+                className="fa fa-plus-circle"
+                onClick={this.addYValue}
+              />
+            </div>
+          </div>
+        </div>
+      );
+    }
+
+    return (
+      <div className="graph-content">
+        <div className="graph-diagram">
+          {chart}
+        </div>
+
+        <div className="x-axis-selector">
+          <h5>X Axis</h5>
+          <select
```

Comment:
This could be replaced with `SelectWithOptions` It should do exactly the same I believe.